### PR TITLE
refactor/726

### LINKED ISSUE #726: Create singleton to store pan and scale values
The `pan` values currently live within the `CanvasController` and the `scale` value lives within the `UiController`.  This is confusing and should be consolidated.

These values should live within their own class which would provide getters and setters and some translation methods.  The classes using these values can then reference the singleton as a single source of truth.

should store:
- `canvas.size.width` and `canvas.size.height`
- `panX` and `panY`
- `scale`

### PR DESCRIPTION
**Pull request by n8rzz**
Resolves #726 

Moves canvas scale, pan and size properties to a singleton with additional translation methods

_originally branched from #725_

### CODE REVIEW

**Comment by n8rzz** [comment]
@erikquinn this is nearly there.  I just have documentation and changelog to add but the logic is done.  feel free to review when you have time.

File: src/assets/scripts/client/constants/canvasConstants.js [hunk 1]
```diff
@@ -1,16 +1,36 @@
...
+    FOTTER_HEIGHT_OFFSET: 36
+};
+
+/**
+ * @enum SCALE
+ * @final
+ */
+export const SCALE = {
+    // FIXME: not a fan of this name
+    CHANGE_FACTOR: 0.9,
```
Comment (erikquinn) [inline src/assets/scripts/client/constants/canvasConstants.js]: `INCREMENT` or `SCROLL_INCREMENT`?

File: src/assets/scripts/client/InputController.js [hunk 2]
```diff
@@ -281,14 +286,15 @@ export default class InputController {
...
+                CanvasStageModel.translatePixelsToKilometers(currentMousePosition[1])
             ];
 
-            position[0] = parseFloat(position[0].toFixed(2));
-            position[1] = parseFloat(position[1].toFixed(2));
-            this.input.positions += `[${position.join(',')}]`;
+            currentMousePosition[0] = parseFloat(currentMousePosition[0].toFixed(2));
+            currentMousePosition[1] = parseFloat(currentMousePosition[1].toFixed(2));
+            // FIXME: what the is this?!
+            this.input.positions += `[${currentMousePosition.join(',')}]`;
```
Comment (erikquinn) [inline src/assets/scripts/client/InputController.js]: I guess it's storing the positions clicked? No idea why; likely to be _extremely_ old.

**Comment by n8rzz** [comment]
@erikquinn what, exactly, is said funny business?  I can't tell from the screen shot.

**Comment by erikquinn** [comment]
Just pan around and you'll see. The position and history dots will move incorrectly; seem to have a Y axis position where the offset is being applied in the wrong direction (ie flip a sign somewhere).

**Comment by erikquinn** [comment]
Perhaps something in the `cc.arc()` in `CanvasController.canvas_draw_radar_targets()`:
```
cc.arc(
    UiController.km_to_px(position[0]) + this.canvas.panX,
    UiController.km_to_px(-position[1]) + this.canvas.panY,
    UiController.km_to_px(this.theme.RADAR_TARGET.HISTORY_DOT_RADIUS_KM),
    0,
    tau()
);
```

**Comment by n8rzz** [comment]
@erikquinn found it, fixed it and ready for review

**Pull request merged**